tar/storage/storage_read.c

```diff
@@ -549,18 +549,20 @@ callback_read_file_response(void * cookie, NETPACKET_CONNECTION * NPC,
 		}
 
 		/* Should we cache this data? */
-		classname[0] = C->class;
-		memcpy(&classname[1], C->name, 32);
-		if (((CF = rwhashtab_read(C->S->cache_ht, classname)) != NULL) &&
-		    (CF->inqueue != 0) && (CF->buf == NULL) && (sc == 0)) {
-			/* Make a copy of this buffer if we can. */
-			if ((CF->buf = malloc(C->buflen)) != NULL) {
-				/* Copy in data and data length. */
-				CF->buflen = C->buflen;
-				memcpy(CF->buf, C->buf, C->buflen);
-
-				/* We've got more data cached now. */
-				C->S->cachesz += CF->buflen;
+		if (sc == 0) {
```

Comment:
How about we drop the `if (sc == 0)` and make this `case 0:` of the preceding `switch` statement?  That seems pretty self-explanatory if we successfully decrypted the data, check if we want to stuff it into the hash table.

Reply:
That would also make it possible to simplify the series of `if` conditions by replacing with a series of (commented) `if (blah) break;` statements covering all the reasons we might *not* put data into the hash table.

Reply:
Yes to using `case 0:` instead of `if (sc == 0)`.

We can't do the `if (blah) break;` if this code is in a separate function (and later, separate file).  But I've rewritten that function to use `if (blah) return;`, which is the same idea.